File: Abeyonix_Frontend/src/types/shop.ts
export interface ShopProductItem {
  id: number;
  name: string;
  slug: string;
  sku: string;

  category_id: number;
  category_name: string;

  sub_category_id?: number | null;
  sub_category_name?: string | null;

  brand?: string | null;

  price?: number | null;
  discount_price?: number | null;

  primary_image?: string | null;
}

export interface ShopProductScrollResponse {
  items: ShopProductItem[];
  last_id: number | null;
  has_more: boolean;
}

export interface GetShopProductsParams {
  category_id?: number;
  sub_category_id?: number;
  last_id?: number;
  limit?: number;
}

export interface ProductSearchItem {
  id: number;
  name: string;
  primary_image?: string;
  price: number;
}

export interface ProductSearchResponse {
  next_cursor?: number | null;
  items: ProductSearchItem[];
}

File: Abeyonix_Backend/app/schemas/shop.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime



class SubCategoryTreeResponse(BaseModel):
    id: int
    name: str
    slug: str
    image_path: Optional[str]
    category_name: Optional[str]

    class Config:
        from_attributes = True


class CategoryTreeResponse(BaseModel):
    id: int
    name: str
    slug: str
    image_path: Optional[str]
    children: List["CategoryTreeResponse"] = []
    sub_categories: List[SubCategoryTreeResponse] = []

    class Config:
        from_attributes = True


CategoryTreeResponse.model_rebuild()



# ----------------------------------------------


class ShopProductResponse(BaseModel):
    id: int
    name: str
    slug: str
    sku: str

    category_id: int
    category_name: Optional[str]

    sub_category_id: Optional[int]
    sub_category_name: Optional[str]

    brand: Optional[str]

    price: Optional[float]
    discount_price: Optional[float]

    primary_image: Optional[str]

    class Config:
        from_attributes = True


class ShopProductScrollResponse(BaseModel):
    items: List[ShopProductResponse]
    last_id: Optional[int]
    has_more: bool


class ProductSearchItem(BaseModel):
    id: int
    name: str
    primary_image: Optional[str]
    price: float

    class Config:
        from_attributes = True


class ProductSearchResponse(BaseModel):
    next_cursor: Optional[int]
    items: List[ProductSearchItem]